src/ssl.ml

```diff
@@ -265,91 +263,152 @@ external set_host : socket -> string -> unit = "ocaml_ssl_set1_host"
+    let sock =
+      Unix.socket domain Unix.SOCK_STREAM 0 in
     try
       Unix.connect sock sockaddr;
       let ssl = embed_socket sock context in
-        connect ssl; ssl
+      connect ssl; ssl
     with
-      | exn -> Unix.close sock; raise exn
+    | exn -> Unix.close sock; raise exn
 
-let open_connection ssl_method sockaddr =
-  open_connection_with_context (create_context ssl_method Client_context) sockaddr
+  let open_connection ssl_method sockaddr =
+    open_connection_with_context (create_context ssl_method Client_context) sockaddr
 
-let close_notify = ssl_shutdown
 
-let shutdown sock =
-  if not (close_notify sock)
-  then ignore (close_notify sock : bool)
+  let close_notify = ssl_shutdown
 
-let shutdown_connection = shutdown
+  let shutdown sock =
+    if not (close_notify sock)
+    then ignore (close_notify sock : bool)
 
-let output_string ssl s =
-  ignore (write_substring ssl s 0 (String.length s))
+  let shutdown_connection = shutdown
 
-let output_char ssl c =
-  let tmp = String.make 1 c in
+  let output_string ssl s =
+    ignore (write_substring ssl s 0 (String.length s))
+
+  let output_char ssl c =
+    let tmp = String.make 1 c in
     ignore (write_substring ssl tmp 0 1)
 
-let output_int ssl i =
-  let tmp = Bytes.create 4 in
+  let output_int ssl i =
+    let tmp = Bytes.create 4 in
     Bytes.set tmp 0 (char_of_int (i lsr 24));
     Bytes.set tmp 1 (char_of_int ((i lsr 16) land 0xff));
     Bytes.set tmp 2 (char_of_int ((i lsr 8) land 0xff));
     Bytes.set tmp 3 (char_of_int (i land 0xff));
-    if write ssl tmp 0 4 <> 4 then failwith "output_int error: all the byte were not sent"
-
-let input_string ssl =
-  let bufsize = 1024 in
-  let buf = Bytes.create bufsize in
-  let ret = ref "" in
-  let r = ref 1 in
+    if write ssl tmp 0 4 <> 4 then
+      failwith "output_int error: all the byte were not sent"
+
+  let input_string ssl =
+    let bufsize = 1024 in
+    let buf = Bytes.create bufsize in
+    let ret = ref "" in
+    let r = ref 1 in
     while !r <> 0
     do
       r := read ssl buf 0 bufsize;
       ret := !ret ^ (Bytes.sub_string buf 0 !r)
     done;
     !ret
 
-let input_char ssl =
-  let tmp = Bytes.create 1 in
+  let input_char ssl =
+    let tmp = Bytes.create 1 in
     if read ssl tmp 0 1 <> 1 then
       raise End_of_file
     else
       Bytes.get tmp 0
 
-let input_int ssl =
-  let i = ref 0 in
-  let tmp = Bytes.create 4 in
+  let input_int ssl =
+    let i = ref 0 in
+    let tmp = Bytes.create 4 in
     ignore (read ssl tmp 0 4);
     i := int_of_char (Bytes.get tmp 0);
     i := (!i lsl 8) + int_of_char (Bytes.get tmp 1);
     i := (!i lsl 8) + int_of_char (Bytes.get tmp 2);
     i := (!i lsl 8) + int_of_char (Bytes.get tmp 3);
     !i
+
+end
+
+(* We apply the functor twice *)
+module SslRelease = SslCom(SslReleaseBase)
+module SslNoRelease = SslCom(SslNoReleaseBase)
+
+(* The releasing functions are imported as default *)
+include SslRelease
```

Comment:
we can save a line here by applying the functor inline:

```ocaml
include Make_SSL (SslReleaseBase)
```

Reply:
Addressed by [28ad4b5](https://github.com/savonet/ocaml-ssl/pull/106/commits/28ad4b5b6aaaca38093c37404ab95fc4cfae4ca6)